### PR TITLE
feat(UI): Implement top bottom widget

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements the top bottom widget.

https://github.com/centreon/centreon/assets/12515407/e2f13014-1e35-4610-b768-826e8da00681

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Noha-ElAbrouki — DISMISSED** [review]

File: centreon/cypress/visual-testing-snapshots/www/widgets/src/centreon-widget-singlemetric/src/SingleMetric.cypress.spec.tsx/Single metric Widget -- Single bar -- displays the metric value as critical when the critical threshold is customized.snap.png
Comment (SoldierBoy31) [inline centreon/cypress/visual-testing-snapshots/www/widgets/src/centreon-widget-singlemetric/src/SingleMetric.cypress.spec.tsx/Single metric Widget -- Single bar -- displays the metric value as critical when the critical threshold is customized.snap.png:1]: Is it normal that the component is not centered on the page ?

Comment (Thebarda) [reply centreon/cypress/visual-testing-snapshots/www/widgets/src/centreon-widget-singlemetric/src/SingleMetric.cypress.spec.tsx/Single metric Widget -- Single bar -- displays the metric value as critical when the critical threshold is customized.snap.png:1]: Yes it's normal bc I haven't defined the width of the container which is defined in the widget

**Review by SoldierBoy31 — APPROVED** [review]

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -15961,7 +15961,7 @@ msgstr ""
 # msgid "Business Views events"
 # msgstr ""
 
-# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
+# msgid "By turning off this option, all links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15964]:
```suggestion
# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16295,7 +16295,7 @@ msgstr ""
 # msgid "Business Views events"
 # msgstr ""
 
-# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
+# msgid "By turning off this option, all links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16298]:
```suggestion
# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -18028,8 +18028,8 @@ msgstr "Afficher la description"
 msgid "Open links in new tab"
 msgstr "Ouvrir les liens dans un nouvel onglet"
 
-msgid "By turning off this option, all links included in the widget will be opened in the same tab."
-msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet."
+msgid "By turning off this option, all links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18031]:
```suggestion
msgid "By turning off this option, all links included in the widget will be opened in the same tab."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -18028,8 +18028,8 @@ msgstr "Afficher la description"
...
+msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18032]:
```suggestion
msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -18058,8 +18058,8 @@ msgstr "Votre widget a été créé avec succès !"
 msgid "Value settings"
 msgstr "Paramètres de valeur"
 
-msgid "By turning off this option, all the links included in the widget will be opened in the same tab."
-msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet."
+msgid "By turning off this option, all the links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18061]:
```suggestion
msgid "By turning off this option, all the links included in the widget will be opened in the same tab."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -18058,8 +18058,8 @@ msgstr "Votre widget a été créé avec succès !"
...
+msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18062]:
```suggestion
msgstr "En désactivant cette option, tous les liens inclus dans le widget seront ouverts dans le même onglet."
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16760,7 +16760,7 @@ msgstr ""
 # msgid "Business Views events"
 # msgstr ""
 
-# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
+# msgid "By turning off this option, all links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:16763]:
```suggestion
# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16749,7 +16749,7 @@ msgstr "Você não salvou as alterações, deseja continuar?"
 # msgid "Business Views events"
 # msgstr ""
 
-# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
+# msgid "By turning off this option, all links included in the widget will be opened in the same tab"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:16752]:
```suggestion
# msgid "By turning off this option, all links included in the widget will be opened in the same tab."
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by SoldierBoy31 — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**